src/Common/Commands.Common.Test/TestClient.cs

```diff
@@ -3,36 +3,31 @@
  * Licensed under the MIT License.
  */
 
-using System;
-using System.Collections.Generic;
-using System.Text;
-using Microsoft.PowerBI.Api.V2;
 using Microsoft.PowerBI.Common.Abstractions.Interfaces;
+using Microsoft.PowerBI.Common.Api;
 using Microsoft.PowerBI.Common.Client;
-using Microsoft.Rest;
 
 namespace Microsoft.PowerBI.Commands.Common.Test
 {
     public class TestClient : IPowerBIClientFactory
     {
+        public IPowerBIApiClient Client { get; }
+
         public FakeHttpClientHandler ClientHandler { get; }
 
+        public TestClient(IPowerBIApiClient client)
+        {
+            this.Client = client;
+        }
+
         public TestClient(FakeHttpClientHandler handler)
         {
             this.ClientHandler = handler;
         }
 
-        public IPowerBIClient CreateClient(IAuthenticationFactory authenticator, IPowerBIProfile profile, IPowerBILogger logger, IPowerBISettings settings)
+        public IPowerBIApiClient CreateClient(IAuthenticationFactory authenticator, IPowerBIProfile profile, IPowerBILogger logger, IPowerBISettings settings)
         {
-            var token = authenticator.Authenticate(profile, logger, settings);
-            if (Uri.TryCreate(profile.Environment.GlobalServiceEndpoint, UriKind.Absolute, out Uri baseUri))
-            {
-                return new PowerBIClient(baseUri, new TokenCredentials(token.AccessToken), this.ClientHandler);
-            }
-            else
-            {
-                return new PowerBIClient(new TokenCredentials(token.AccessToken), this.ClientHandler);
-            }
+            return this.Client ?? new PowerBIApiClient(authenticator, profile, logger, settings, this.ClientHandler);
```

Comment:
Should we just throw an exception here instead of creating the real PowerBIApiClient?